### PR TITLE
docs(readme): update `fastify-auth` example

### LINKED ISSUE #80: fastify-auth sample seems not to work correctly
## 🐛 Bug Report

Using the delivered example for fastify-auth integration gives me an error: 
```js
(node:19472) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'bind' of undefined
    at Object.auth (E:\Files\Projects\Node.JS\Test\BetaLoad\masterapi\node_modules\fastify-auth\auth.js:33:33)
    at main (E:\Files\Projects\Node.JS\Test\BetaLoad\masterapi\main.js:17:25)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:19472) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:19472) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

## To Reproduce

I used the example code as a base. Registering fastify-auth first with fastify. Then fastify-bearer-auth. I had to do this with await otherwise i got an error that app.auth() is not known.

Full code:
```js
'use strict'

const app = require('fastify')({
    logger: true
});
const bearerAuthPlugin = require('fastify-bearer-auth');
const fastifyAuth = require('fastify-auth');
const keys = new Set(['a-super-secret-key', 'another-super-secret-key']);

async function main() {
    await app.register(fastifyAuth);
    await app.register(bearerAuthPlugin, { addHook: false, keys});

    app.route({
        method: 'GET',
        url: '/multiauth',
        preHandler: app.auth([
            app.allowAnonymous,
            app.verifyBearerAuth
        ]),
        handler: function (_, reply) {
            reply.send({ hello: 'world' })
        }
    });

    app.listen({port: 3000}, (err, address) => {
        if (err) {
            app.log.error(err.message)
            process.exit(1)
        }
        app.log.info(`server listening on ${address}`);
    });
}

main();
```

## Expected behavior

Code should run.

## Your Environment

- *node version*: 12
- *fastify version*: 3.14.2
- *os*: Windows
- *fastify-auth version*: 1.1.0

### PR DESCRIPTION
**Pull request by Fdawgs**
closes #80 by resolving the following issues with the original example:

- `fastify-auth` plugin not registered
- `allowAnonymous` decorator not added
- `fastify-auth` and `fastify-bearer-auth` plugins not waited on, leading to `UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'bind' of undefined`  being thrown

#### Checklist

- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -91,22 +91,44 @@ This plugin can integrate with `fastify-auth` by following this example:
...
+    preHandler: fastify.auth([
+      fastify.allowAnonymous,
+      fastify.verifyBearerAuth
+    ]),
+    handler: function (_, reply) {
+      reply.send({ hello: 'world' })
+    }
+  })
+
+  fastify.listen({port: 8000}, (err) => {
```
Comment (Eomm) [inline Readme.md:122]: Could we use await here to avoid mixing too many patterns?

Comment (Fdawgs) [reply Readme.md]: Updated in https://github.com/fastify/fastify-bearer-auth/pull/102/commits/3845b7e3e3f52f0ce8291f0817766d8c8e5bb3d7

Comment (Eomm) [reply Readme.md]: If there is a callback, a promise is not returned 😞

File: Readme.md [hunk 2]
```diff
@@ -91,22 +91,44 @@ This plugin can integrate with `fastify-auth` by following this example:
...
-  url: '/multiauth',
-  preHandler: fastify.auth([
-    fastify.allowAnonymous,
-    fastify.verifyBearerAuth
-  ]),
-  handler: function (_, reply) {
-    reply.send({ hello: 'world' })
-  }
-})
+const server = async function () {
```
Comment (jsumners) [inline Readme.md:98]: This function is returning `undefined`.

**Review by Eomm — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**